0.9.3 unable to use only JGit

When there is no git in the system, then (useJGit enabled):
```
java.io.IOException: Cannot run program "git": error=2, Нет такого файла или каталога
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at scala.sys.process.ProcessBuilderImpl$Simple.run(ProcessBuilderImpl.scala:68)
	at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.lines(ProcessBuilderImpl.scala:140)
	at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.lines_$bang(ProcessBuilderImpl.scala:108)
	at com.typesafe.sbt.SbtGit$$anonfun$buildSettings$11.apply(SbtGit.scala:138)
	at com.typesafe.sbt.SbtGit$$anonfun$buildSettings$11.apply(SbtGit.scala:122)
	at sbt.Init$Value$$anonfun$apply$13.apply(Settings.scala:609)
	at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$constant$1.apply(INode.scala:163)
	at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$constant$1.apply(INode.scala:163)
	at sbt.EvaluateSettings$MixedNode.evaluate0(INode.scala:175)
	at sbt.EvaluateSettings$INode.evaluate(INode.scala:135)
	at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$submitEvaluate$1.apply$mcV$sp(INode.scala:69)
	at sbt.EvaluateSettings.sbt$EvaluateSettings$$run0(INode.scala:78)
	at sbt.EvaluateSettings$$anon$3.run(INode.scala:74)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: error=2, Нет такого файла или каталога
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	at scala.sys.process.ProcessBuilderImpl$Simple.run(ProcessBuilderImpl.scala:68)
	at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.lines(ProcessBuilderImpl.scala:140)
	at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.lines_$bang(ProcessBuilderImpl.scala:108)
	at com.typesafe.sbt.SbtGit$$anonfun$buildSettings$11.apply(SbtGit.scala:138)
	at com.typesafe.sbt.SbtGit$$anonfun$buildSettings$11.apply(SbtGit.scala:122)
	at sbt.Init$Value$$anonfun$apply$13.apply(Settings.scala:609)
	at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$constant$1.apply(INode.scala:163)
	at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$constant$1.apply(INode.scala:163)
	at sbt.EvaluateSettings$MixedNode.evaluate0(INode.scala:175)
	at sbt.EvaluateSettings$INode.evaluate(INode.scala:135)
	at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$submitEvaluate$1.apply$mcV$sp(INode.scala:69)
	at sbt.EvaluateSettings.sbt$EvaluateSettings$$run0(INode.scala:78)
	at sbt.EvaluateSettings$$anon$3.run(INode.scala:74)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/src/main/scala/com/typesafe/sbt/SbtGit.scala
+++ b/src/main/scala/com/typesafe/sbt/SbtGit.scala
@@ -135,11 +135,11 @@ object SbtGit {
         )
       )
 
-      Process("git ls-remote --get-url origin").lines_!.head match {
+      gitReader.value.withGit(_.remoteOrigin) match {
         case unauthenticated(domain, repo) => buildScmInfo(domain,repo)
         case ssh(domain, repo) => buildScmInfo(domain,repo)
         case headlessSSH(domain, repo) => buildScmInfo(domain,repo)
-	case _ => None
+	      case _ => None
       }
     }
   )

--- a/src/main/scala/com/typesafe/sbt/SbtGit.scala
+++ b/src/main/scala/com/typesafe/sbt/SbtGit.scala
@@ -139,7 +139,7 @@ object SbtGit {
         case unauthenticated(domain, repo) => buildScmInfo(domain,repo)
         case ssh(domain, repo) => buildScmInfo(domain,repo)
         case headlessSSH(domain, repo) => buildScmInfo(domain,repo)
-	      case _ => None
+        case _ => None
       }
     }
   )

--- a/src/main/scala/com/typesafe/sbt/git/JGit.scala
+++ b/src/main/scala/com/typesafe/sbt/git/JGit.scala
@@ -95,6 +95,17 @@ final class JGit(val repo: Repository) extends GitReadonlyInterface {
     porcelain.branchList.setListMode(ListMode.REMOTE).call.asScala.filter(_.getName.startsWith("refs/remotes")).map(_.getName.drop(13))
   }
 
+  override def remoteOrigin: String = {
+    // same functionality as Process("git ls-remote --get-url origin").lines_!.head
+    import collection.JavaConverters._
+    porcelain.remoteList().call.asScala
+      .filter(_.getName == "origin")
+      .flatMap(_.getURIs.asScala)
+      .headOption
+      .map(_.toString)
+      .getOrElse("origin")
+  }
+
   override def headCommitMessage: Option[String] = Try(Option(porcelain.log().setMaxCount(1).call().iterator().next().getFullMessage)).toOption.flatten
 
   override def headCommitDate: Option[String] = {

--- a/src/main/scala/com/typesafe/sbt/git/ReadableGit.scala
+++ b/src/main/scala/com/typesafe/sbt/git/ReadableGit.scala
@@ -23,6 +23,8 @@ trait GitReadonlyInterface {
   def branches : Seq[String]
   /** The remote branches */
   def remoteBranches: Seq[String]
+  /** The remote origin as returned by `git ls-remote --get-url origin`. */
+  def remoteOrigin: String
   /** The message of current commit **/
   def headCommitMessage: Option[String]
 }

